### PR TITLE
demos: add no_show for C++, update tests

### PR DESCRIPTION
**Pull request by Wovchena**

### CODE REVIEW

File: demos/object_detection_demo_ssd_async/README.md [hunk 1]
```diff
@@ -112,14 +112,15 @@ Options:
     -h                        Print a usage message.
     -i "<path>"               Required. Path to a video file (specify "cam" to work with camera).
     -m "<path>"               Required. Path to an .xml file with a trained model.
-      -l "<absolute_path>"    Required for CPU custom layers. Absolute path to a shared library with the kernel implementations.
+      -l "<absolute_path>"    Required for CPU custom layers. Absolute path to a shared library with the kernels implementations.
```
Comment (IRDonch) [inline demos/object_detection_demo_ssd_async/README.md]: This is a) grammatically incorrect and b) unrelated to this PR.

File: demos/object_detection_demo_ssd_async/main.cpp [hunk 2]
```diff
@@ -319,7 +319,6 @@ int main(int argc, char *argv[]) {
                 for (int i = 0; i < maxProposalCount; i++) {
                     float image_id = detections[i * objectSize + 0];
                     if (image_id < 0) {
-                        std::cout << "Only " << i << " proposals found" << std::endl;
```
Comment (IRDonch) [inline demos/object_detection_demo_ssd_async/main.cpp]: What's the purpose of this?

File: demos/object_detection_demo_ssd_async/models.lst [hunk 3]
```diff
@@ -8,3 +8,4 @@ pedestrian-detection-adas-binary-????
 vehicle-detection-adas-????
 vehicle-detection-adas-binary-????
 vehicle-license-plate-detection-barrier-????
+person-detection-retail-????
```
Comment (IRDonch) [inline demos/object_detection_demo_ssd_async/models.lst]: Please keep these in sorter order.

File: demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp [hunk 4]
```diff
@@ -43,6 +43,8 @@ static const char raw_output_message[] = "Optional. Inference results as raw val
 /// @brief message resizable input flag
 static const char input_resizable_message[] = "Optional. Enables resizable input with support of ROI crop & auto resize.";
 
+/// @brief Message do not show processed video
+static const char no_show_processed_video[] = "Optional. Do not show processed video.";
```
Comment (IRDonch) [inline demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp]: Shouldn't this be called `no_show_message`, for consistency?

File: demos/object_detection_demo_ssd_async/README.md (hunk 1 above)
Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/README.md]: Reverted

File: demos/object_detection_demo_ssd_async/main.cpp (hunk 2 above)
Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/main.cpp]: This print is not really useful and buggy (If there are exactly `maxProposalCount` detections, the print will not be triggered)

File: demos/object_detection_demo_ssd_async/models.lst (hunk 3 above)
Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/models.lst]: Updated

File: demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp (hunk 4 above)
Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp]: I agree, that `no_show_message` would be clearer, but `text_detection_demo` only names it as `no_show_message`, and most demos use `no_show_processed_video`.

File: demos/object_detection_demo_ssd_async/README.md (hunk 1 above)
Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/README.md]: Submitted https://github.com/opencv/open_model_zoo/pull/386

File: demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp (hunk 4 above)
Comment (vladimir-dudnik) [reply demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp]: so, variable name is no_show_processed_video even if sample do process single image? Or in possible future, input of sample might not be image or video at all. It seems make sense to change name of variable consistently and input data type agnostic for all demos. Might be make this as a separate PR?

Comment (Wovchena) [reply demos/object_detection_demo_ssd_async/object_detection_demo_ssd_async.hpp]: Submitted the issue, that generalizes this comment: https://github.com/opencv/open_model_zoo/issues/391

**Pull request merged**